cmd/performance-profile-creator/cmd/root.go

```diff
@@ -0,0 +1,109 @@
+)
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "performance-profile-creator",
+	Short: "A tool that automates creation of Performance Profiles",
+	Run:   func(cmd *cobra.Command, args []string) {},
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+	rootCmd.PersistentFlags().IntVarP(&reservedCPUCount, "reserved-cpu-count", "R", 0, "Number of reserved CPUs")
+	rootCmd.PersistentFlags().BoolVarP(&splitCPUsAcrossNUMA, "split-cpus-across-numa", "S", true, "Split the CPUs across NUMA nodes")
+	rootCmd.PersistentFlags().BoolVarP(&disableHT, "disable-ht", "H", false, "Disable Hyperthreading")
+	rootCmd.PersistentFlags().BoolVarP(&rtKernel, "rt-kernel", "K", true, "Enable Real Time Kernel")
+	rootCmd.PersistentFlags().BoolVarP(&userLevelNetworking, "user-level-networking", "U", false, "Run with User level Networking(DPDK) enabled")
+	rootCmd.PersistentFlags().StringVarP(&powerConsumptionMode, "power-consumption-mode", "P", "cstate", "The power consumption mode")
+	rootCmd.PersistentFlags().StringVarP(&outputFilePath, "output-file-path", "O", "/output/sample-profile.json", "Performance Profile file output path")
+	rootCmd.PersistentFlags().StringVarP(&mustGatherDirPath, "must-gather-dir-path", "M", "/must-gather", "Must gather directory path")
+	rootCmd.PersistentFlags().StringVarP(&profileName, "profile-name", "N", "performance", "Name of the performance profile to be created")
+
+	// TODO: Input validation
+	// 1) Make flags required/optional
+	// 2) e.g.check to make sure that power consumption string is in {CSTATE NO_CSTATE IDLE_POLL}
+
+	// Creating Performance Profile
+	createProfile(outputFilePath)
+}
+
+func createProfile(outputFilePath string) {
+
+	// TODO: Get the name from MCP if not specified in the command line arguments
+	profile := &performancev2.PerformanceProfile{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PerformanceProfile",
+			APIVersion: performancev2.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: profileName,
+		},
+		Spec: performancev2.PerformanceProfileSpec{
+			RealTimeKernel: &performancev2.RealTimeKernel{
+				Enabled: pointer.BoolPtr(true),
+			},
+			AdditionalKernelArgs: []string{},
+			NUMA: &performancev2.NUMA{
+				TopologyPolicy: pointer.StringPtr("restricted"),
+			},
+		},
+	}
+
+	var performanceProfileData []byte
+	var err error
+
+	if performanceProfileData, err = json.Marshal(&profile); err != nil {
+		klog.Errorf("Unable to Marshal sample performance profile: %v", err)
+	}
+
+	klog.Infof("The performance profile data is %q", performanceProfileData)
+
+	err = ioutil.WriteFile(outputFilePath, performanceProfileData, 0777)
```

Comment:
0777 seems too much :) I'd start with 0644

Reply:
+1

Reply:
Sure, will fix this!

Reply:
Well, now that I think again, we are planning to send the data just to STDOUT and hence we won't need to write to the file.